framework/encode/vulkan_state_writer.cpp

```diff
@@ -4338,5 +4340,583 @@ void VulkanStateWriter::WriteExecuteFromFile(const std::string& filename, uint32
     blocks_written_ += n_blocks + 1;
 }
 
+void VulkanStateWriter::WriteDebugUtilsState(const VulkanStateTable& state_table)
+{
+    // state_table.VisitWrappers([&](const vulkan_wrappers::BufferWrapper* wrapper) {})
+    state_table.VisitWrappers([&](const vulkan_wrappers::InstanceWrapper* wrapper) {
```

Comment:
Is every one of these visits the same code?  Can the body be made into a template lambda that is then called in the passed lambda?  E.g. something like
```
    auto write_debug_utils_calls = [&](const auto *wrapper) {
        assert(wrapper != nullptr);
        if (wrapper->debug_name_create_parameters)
        {
            WriteFunctionCall(format::ApiCall_vkSetDebugUtilsObjectNameEXT, wrapper->debug_name_create_parameters.get());
        }

        if (wrapper->debug_tag_create_parameters)
        {
            WriteFunctionCall(format::ApiCall_vkSetDebugUtilsObjectTagEXT, wrapper->debug_tag_create_parameters.get());
        }
    };

    state_table.VisitWrappers([&](const vulkan_wrappers::InstanceWrapper* wrapper) { write_debug_utils_calls(wrapper); });
    state_table.VisitWrappers([&](const vulkan_wrappers::PhysicalDeviceWrapper* wrapper) { write_debug_utils_calls(wrapper); });
    ...
```

Reply:
done